File: app/src/main/java/co/medellinandroid/spotifyartists/topTenArtists/TopTenArtistsContract.java
package co.medellinandroid.spotifyartists.topTenArtists;

import co.medellinandroid.spotifyartists.model.SpotifyArtistsItems;
import java.util.List;

public interface TopTenArtistsContract {

  interface View{
    void showProgress();
    void hideProgress();
    void showError(String message);
    void showList(List<SpotifyArtistsItems> topTenArtists);
  }

  interface Presenter{
    void attachView(View view);
    void getTopTenArtists();
    void onDestroy();
  }
}

File: app/src/main/java/co/medellinandroid/spotifyartists/topTenArtists/TopTenArtistsContract.kt
package co.medellinandroid.spotifyartists.topTenArtists

import co.medellinandroid.spotifyartists.model.SpotifyArtistsItems

interface TopTenArtistsContract {

  interface View {
    fun showProgress()
    fun hideProgress()
    fun showError(message: String)
    fun showList(topTenArtists: List<SpotifyArtistsItems>)
  }

  interface Presenter {
    fun attachView(view: View)
    fun getTopTenArtists()
    fun onDestroy()
  }
}
